File: app/src/main/java/com/jdappel/beerinvestigator/AppComponent.java
package com.jdappel.beerinvestigator;

import com.jdappel.beerinvestigator.BeerApplication;
import com.jdappel.beerinvestigator.rest.BreweryDBServiceModule;
import com.jdappel.beerinvestigator.ui.beers.ActivityModule;
import com.jdappel.beerinvestigator.ui.viewmodel.impl.BeerModule;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton
@Component(modules = {BreweryDBServiceModule.class, AndroidInjectionModule.class, ActivityModule.class, BeerModule.class})
interface AppComponent {
    void inject(BeerApplication application);
}

File: app/src/main/java/com/jdappel/beerinvestigator/AppComponent.kt
package com.jdappel.beerinvestigator

import com.jdappel.beerinvestigator.di.NetworkModule
import com.jdappel.beerinvestigator.di.ViewModelModule
import com.jdappel.beerinvestigator.data.repo.RepoModule
import javax.inject.Singleton
import com.jdappel.beerinvestigator.di.UtilsModule
import dagger.android.AndroidInjectionModule
import com.jdappel.beerinvestigator.ui.beers.ActivityModule
import com.jdappel.beerinvestigator.ui.viewmodel.impl.BeerModule
import dagger.Component

@Singleton
@Component(modules = [ViewModelModule::class, NetworkModule::class, AndroidInjectionModule::class, ActivityModule::class, BeerModule::class, RepoModule::class, UtilsModule::class])
interface AppComponent {
    fun inject(application: BeerApplication?)
}